CAP/gap/MethodRecord.gi

```diff
@@ -2952,6 +2952,52 @@ SolveLinearSystemInAbCategory := rec(
   return_type := "morphism_or_fail"
 ),
 
+RandomObject := rec(
+  installation_name := "RandomObject",
+  filter_list := [ "category", IsInt ],
+  return_type := "object"
+),
+
+RandomMorphism := rec(
+  installation_name := "RandomMorphism",
+  filter_list := [ "category", IsInt ],
+  return_type := "morphism"
+),
+
+RandomMorphismWithFixedSource := rec(
+  installation_name := "RandomMorphismWithFixedSource",
+  filter_list := [ "object", IsInt ],
+  return_type := "morphism",
+  post_function := function( object, n, return_value )
+    if not IsEqualForObjects( object, Source( return_value ) ) then
+      Error( "The source of the returned morphism is not equal to the given object" );
+    fi;
+  end 
+),
+
+RandomMorphismWithFixedRange := rec(
+  installation_name := "RandomMorphismWithFixedRange",
+  filter_list := [ "object", IsInt ],
+  return_type := "morphism",
+  post_function := function( object, n, return_value )
+    if not IsEqualForObjects( object, Range( return_value ) ) then
+      Error( "The range of the returned morphism is not equal to the given object" );
+    fi;
+  end 
+),
+
+RandomMorphismWithFixedSourceAndRange := rec(
+  installation_name := "RandomMorphismWithFixedRange",
```

Comment:
In the current commits in graded and non-graded case, a function TRY_TO_ENHANCE_HOMALG_(GRADED)_RING_WITH_RANDOM_FUNCTIONS
will be called in the category constructor method.

The output is true if it succeed to add the random functions to the ring and false otherwise and only in the affirmative case the random methods will be added to category.